uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-request-action-dropdown.js

```diff
@@ -0,0 +1,172 @@
+/* eslint-env amd */
+
+define([
+  'common/lodash',
+  'common/moment',
+  'leave-absences/shared/modules/components',
+  'common/services/hr-settings'
+], function (_, moment, components) {
+  components.component('leaveRequestActionDropdown', {
+    bindings: {
+      request: '<',
+      requestStatuses: '<',
+      absenceTypes: '<',
+      /**
+       * Role is not a permission level in this case.
+       * For example Manager can act as Staff
+       * and Admin can act as either Manager or Staff.
+       */
+      role: '<'
+    },
+    templateUrl: ['shared-settings', function (sharedSettings) {
+      return sharedSettings.sharedPathTpl + 'components/leave-request-action-dropdown.html';
+    }],
+    controllerAs: 'vm',
+    controller: ['$log', '$q', '$rootScope', 'OptionGroup', 'dialog', 'shared-settings', controller]
+  });
+
+  function controller ($log, $q, $rootScope, OptionGroup, dialog, sharedSettings) {
+    $log.debug('Component: leave-request-action-dropdown');
+
+    var vm = this;
+
+    vm.actions = [];
```

Comment:
Why don't you have a single object, listing all the actions indexed by key, where you put all the relevant information for each action?
```js
vm.actions = {
  approve: {
    name: 'Approve',
    rolesAllowed: ['manager', 'admin'],
    statusAllowed: [sn.awaitingApproval]
    dialog: {
      title: 'Approval',
      btnClass: 'success',
      btnLabel: 'Approve',
      msg: 'Please confirm approval'
    }
  },
  cancel: {
    name: 'Cancel',
    rolesAllowed: ['staff', 'manager', 'admin'],
    statusAllowed: [sn.awaitingApproval, sn.moreInformationRequired]
    dialog: {
      title: 'Approval',
      btnClass: 'success',
      btnLabel: 'Approve',
      msg: 'Please confirm approval'
    }
  }
};
```
This way you have all the info you need in just one place. And then in `getActions` you simply filter the list by leave request status and role